File: Chapter_1/01.2ndsolution.js
function sortedSquared(array){
    const newArray = new Array(array.length).fill(0)
    let leftPointer = 0
    let rightPointer = array.length-1;
    for(let i =array.length-1;i>=0;i--){
        const leftSquared = Math.pow(array[leftPointer],2);
        const rightSquared = Math.pow(array[rightPointer],2);
        if(leftSquared>rightSquared){
            newArray[i]= leftSquared;
            leftPointer++;
        } else{
            newArray[i]= rightSquared;
            rightPointer--;
        }
    }
    return newArray
}

console.log(sortedSquared([-8,4,8,9]))

File: Chapter_1/01.2ndsolution.py
def sorted_squared(array):
    i = 0
    j = len(array)-1
    new_Array = [0]*len(array)
    for K in reversed(range(len(array))):
        sq_i = array[i]**2
        sq_j = array[j]**2
        if sq_i>sq_j:
            new_Array[K] = sq_i
            i+=1
        else:
            new_Array[K]= sq_j
            j-=1
    return new_Array


print(sorted_squared([-8,4,8,9]))  